src/unix.rs

```diff
@@ -1,57 +1,30 @@
-use alloc::string::String;
-use core::ptr::NonNull;
-use cstr_core::CStr;
+use std::env;
 
+const LC_ALL: &str = "LC_ALL";
 const LC_CTYPE: &str = "LC_CTYPE";
+const LANG: &str = "LANG";
 
 pub(crate) fn get() -> Option<String> {
-    // SAFETY: `setlocale` is being called with a known category and an empty, null-terminated string.
-    let locale = NonNull::new(unsafe { libc::setlocale(libc::LC_ALL, b"\0".as_ptr().cast()) })?;
+    let code = env::var(LC_ALL)
+        .or_else(|_| env::var(LC_CTYPE))
+        .or_else(|_| env::var(LANG))
+        .ok()?;
 
-    // SAFETY: `setlocale` returns a pointer to a NUL terminated string which
-    // is then cloned into an owned Rust string, and has otherwise been checked as non-null.
-    let code = unsafe { CStr::from_ptr(locale.as_ptr()) };
-    let code = code.to_str().ok()?;
-
-    parse_locale_code(code)
+    parse_locale_code(&code)
 }
 
 fn parse_locale_code(code: &str) -> Option<String> {
-    // Support two behaviors:
-    //
-    // - `LC_ALL` and `LANG` are set, so `setlocale` returns a single locale identifier.
-    // - `LANG` is set but `LC_ALL` is not, so `setlocale` returns a list of all identifiers.
-    //
-    // In the first case we have a clear case of which locale to use. In the second, we will fallback to using `LC_CTYPE`
-    // as its the most closely related to character localization.
-    let code = if !code.contains(LC_CTYPE) {
-        code
-    } else {
-        code.split(';')
-            .find(|c| c.starts_with(LC_CTYPE))
-            .and_then(|c| c.split('=').last())?
-    };
-
     // Some locales are returned with the char encoding too: `en_US.UTF-8`
-    code.splitn(2, '.').next().map(String::from)
+    code.split_once('.').map(|(locale, _enc)| locale.to_owned())
```

Comment:
I believe that we will need to use `splitn` here and then tell Clippy to ignore it. The current MSRV is too low to use `split_once` and this feels like a small-enough thing to not bump over.

Reply:
Great catch!